pkg/neorpc/filters.go

```diff
@@ -126,14 +152,71 @@ func (f *NotificationFilter) Copy() *NotificationFilter {
 		res.Name = new(string)
 		*res.Name = *f.Name
 	}
+	if len(f.Parameters) != 0 {
+		res.Parameters = slices.Clone(f.Parameters)
+	}
+	f.parametersCache = nil
 	return res
 }
 
+// ParametersAsStackItems returns [stackitem.Item] version of [NotificationFilter.Parameters]
+// according to [smartcontract.Parameter.ToStackItem]; Notice that the result is cached
+// internally in [NotificationFilter] for efficiency, so once you call this method it will
+// not change even if you change any structure fields. If you need to update parameters, use
+// [NotificationFilter.Copy].It mainly should be used by server code. Must not be used
+// concurrently.
+func (f *NotificationFilter) ParametersAsStackItems() ([]stackitem.Item, error) {
+	if len(f.Parameters) == 0 {
+		return nil, nil
+	}
+	if f.parametersCache != nil {
+		return f.parametersCache, nil
+	}
+	f.parametersCache = make([]stackitem.Item, 0, len(f.Parameters))
+	for i, p := range f.Parameters {
+		si, err := p.ToStackItem()
+		if err != nil {
+			f.parametersCache = nil
+			return nil, fmt.Errorf("converting %d parameter to stack item: %w", i, err)
+		}
+		f.parametersCache = append(f.parametersCache, si)
+	}
+	return f.parametersCache, nil
+}
```

Comment:
And BTW, usually in NeoGo it's vice-versa:
```go
if f.parametersCache == nil {
    // fill parametersCache in
}
return f.parametersCache, nil
```
https://github.com/nspcc-dev/neo-go/blob/ff15e39363ea44cbe337f493c434304de38b2382/pkg/core/transaction/transaction.go#L331-L334

Reply:
it adds additional nesting, usually, i find it a more complex code, but ok